packages/components/src/spinner/index.js

```diff
@@ -1,8 +1,54 @@
+/**
+ * External dependencies
+ */
+import styled from '@emotion/styled';
+
 /**
  * Internal dependencies
  */
-import { StyledSpinner } from './styles/spinner-styles';
+import { COLORS, CONFIG } from '../utils';
+
+/**
+ * WordPress dependencies
+ */
+import { SVG, Circle } from '@wordpress/primitives';
+
+// A dedicated wrapper allows us to apply width and shadow color based on config values
+// The SVG can only have a centered stroke so a stacked box-shadow accomplishes a matching 1.5px border
+export const StyledSpinner = styled.span`
+	width: ${ CONFIG.spinnerSize }px;
+	height: ${ CONFIG.spinnerSize }px;
+	box-shadow: inset 0 0 0 0.75px ${ COLORS.gray[ 300 ] },
+		0 0 0 0.75px ${ COLORS.gray[ 300 ] };
+`;
 
 export default function Spinner() {
-	return <StyledSpinner className="components-spinner" />;
+	const viewBox = `0 0 ${ CONFIG.spinnerSize } ${ CONFIG.spinnerSize }`;
+	const radius = Number( CONFIG.spinnerSize ) / 2;
+	const strokeDasharray = [
+		Number( CONFIG.spinnerSize ),
+		Number( CONFIG.spinnerSize ) * 10,
+	];
+
+	return (
+		<StyledSpinner className="components-spinner">
+			<SVG
+				viewBox={ viewBox }
+				xmlns="http://www.w3.org/2000/svg"
+				className="components-spinner__track"
+				role="img"
+				aria-hidden="true"
+				focusable="false"
+			>
+				<Circle
+					className="components-spinner__indicator"
+					cx="50%"
+					cy="50%"
+					r={ radius }
+					stroke-dasharray={ strokeDasharray }
+					vector-effect="non-scaling-stroke"
```

Comment:
I believe in React this prop should be specific camel-case, i.e. `vectorEffect`

Reply:
Done in https://github.com/WordPress/gutenberg/pull/37551/commits/cd30bdb1fb1c1de37ea37953edf1305425300385, marking as resolved